build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -62,25 +63,42 @@ const currentFile = new Project({
 
 const previousTsConfigPath = `${previousBasePath}/tsconfig.json`;
 let previousFile: SourceFile;
-if (existsSync(previousTsConfigPath)) {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-		tsConfigFilePath: previousTsConfigPath,
-	});
+const project = new Project({
+	skipFileDependencyResolution: true,
+	tsConfigFilePath: existsSync(previousTsConfigPath) ? previousTsConfigPath : undefined,
+});
+// Check for existence of alpha and add appropriate file
+if (existsSync(alphaFilePath)) {
+	project.addSourceFilesAtPaths(alphaFilePath);
+	previousFile = project.getSourceFileOrThrow("<package-name>-alpha.d.ts");
+	// If alpha doesn't exist but index.ts and tsconfig.json do
+} else if (existsSync(previousTsConfigPath)) {
 	previousFile = project.getSourceFileOrThrow("index.ts");
+	// Fall back to using .d.ts
 } else {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-	});
 	project.addSourceFilesAtPaths(`${previousBasePath}/dist/**/*.d.ts`);
 	previousFile = project.getSourceFileOrThrow("index.d.ts");
 }
 
-function typeDataFromFile(file: SourceFile): Map<string, TypeData> {
+/**
+ * Extracts type data from a TS source file and creates a map where each key is a type name and the value is its type data.
+ * @param file
+ * @returns Map<string, TypeData> mapping between item and its type
+ */
+function typeDataFromFile(file: SourceFile, isAlpha: boolean): Map<string, TypeData> {
 	const typeData = new Map<string, TypeData>();
 	const exportedDeclarations = file.getExportedDeclarations();
+
 	for (const declarations of exportedDeclarations.values()) {
 		for (const dec of declarations) {
+			// Check for @alpha tag in block comment if it's an alpha file.
+			if (isAlpha && Node.isJSDocable(dec)) {
```

Comment:
none of this should need to change if you correctly load alpha for the previous package.